backend/OpenAPI.yaml

```diff
@@ -137,37 +119,69 @@ paths:
           content:
             application/json:
               schema:
-                type: object
-                properties:
-                  error:
-                    type: string
-                    example: "Invalid listing ID format. Must be a valid ObjectId."
+                $ref: "#/components/schemas/ErrorResponse"
         "404":
           description: Listing not found
           content:
             application/json:
               schema:
-                type: object
-                properties:
-                  error:
-                    type: string
-                    example: "Listing not found."
+                $ref: "#/components/schemas/ErrorResponse"
+        "422":
+          description: Invalid Input body
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
         "500":
           description: Internal Server Error
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/500ErrorResponse"
+                $ref: "#/components/schemas/ErrorResponse"
 
 components:
   schemas:
-    500ErrorResponse:
+    ErrorResponse:
+      type: object
+      required:
+        - details
+      properties:
+        details:
+          type: string
+      example: 
+        details: "Internal Server Error. Please try again later."
+        # code:
```

Comment:
The code is not part of the response schema since the status code is part of the HTTP header (not the HTTP body).